osmosis-labs/osmosis
# CI: Automatically close broken links issues when new ones are created

Closes: #5806

## What is the purpose of the change

Improves our developer experience by automatically closing old Broken Links issues when we create new ones during our automated run.

Workflow explanation:
- `only-issue-labels: automated issue,Broken links`: To close only issues with those labels (I added the `Broken links` label to the action to make it more unique)
- `close-issue-message: ...`: To have a message explaining the close, and a link to our open issues - if any - with the labels.
    - The token needs the `issues: write` permission for the message to be written. Unsure if it has those permissions.
- `days-before-stale: -1`: To not mark it as stale, ever
- `days-before-close: 0`: To automatically close the issue. 
    - I could not find examples in the `stale` documentation on if this works, but 'logically' it should. This is the main reason I'd want to see about manually running the workflow to check it works.

## Testing and Verifying

There are no tests for the action, but one way to manually verify if this works is to
1. Reopen https://github.com/osmosis-labs/osmosis/issues/5806 with an added `Broken links` label
2. Run the `Check Broken Links` action with my changes
3. View the run and expect no errors
4. Check that the above issue has been closed on run, with the message linking to our PRs with label(s).

Note: It's possible that no new issue gets created if there are no broken links. We can confirm it works as expected by ensuring there are no failures during the CI run. We know the action works since it has created issues before, so the functionality we care about is the new step `Close Existing Broken Links Issue(s)` not breaking the CI proccess.

## Documentation and Release Note

  - [x] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
